### PR TITLE
feat(sdk): type chat.createStartSessionAction against your chat agent

### PR DESCRIPTION
**Pull request by ericallam**
## Summary

Type `chat.createStartSessionAction` against the chat agent so `clientData` is typed end-to-end on the first turn. Closes the gap where `useTriggerChatTransport`'s `startSession` callback already hands you a typed `clientData` (via the transport generic) but the server-side action couldn't accept it without untyped routing through the `metadata` field.

## Design

`ChatStartSessionParams` gains a typed `clientData` field via the new generic:

```ts
export type ChatStartSessionParams<TChat extends AnyTask = AnyTask> = {
  chatId: string;
  clientData?: InferChatClientData<TChat>;
  triggerConfig?: Partial<SessionTriggerConfig>;
  metadata?: Record<string, unknown>;
};

function createChatStartSessionAction<TChat extends AnyTask = AnyTask>(
  taskId: string,
  options?: CreateChatStartSessionActionOptions
): (params: ChatStartSessionParams<TChat>) => Promise<ChatStartSessionResult>
```

When provided, `clientData` is folded into the first run's `triggerConfig.basePayload.metadata`, so `onPreload` / `onChatStart` see the same shape per-turn `metadata` carries via the transport. The opaque session-level `metadata` field stays exactly as before — it lands on the Session row, not the run payload.

## Usage

```ts
// actions.ts
import { chat } from "@trigger.dev/sdk/ai";
import type { myChat } from "@/trigger/chat";

export const startChatSession = chat.createStartSessionAction<typeof myChat>("my-chat");
```

```tsx
// Chat.tsx
const transport = useTriggerChatTransport<typeof myChat>({
  task: "my-chat",
  accessToken: ({ chatId }) => mintChatAccessToken(chatId),
  startSession: ({ chatId, clientData }) =>
    startChatSession({ chatId, clientData }),
});
```

## Test plan

- [x] `pnpm run build --filter @trigger.dev/sdk` passes
- [ ] Verify a `chat.agent` with `clientDataSchema` reads the typed clientData from `onPreload` payload metadata on the first turn

### CODE REVIEW

File: packages/trigger-sdk/src/v3/createStartSessionAction.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    // The agent's typed clientData is strictly narrower than `unknown`.
+    expectTypeOf<Parameters<typeof start>[0]["clientData"]>().not.toEqualTypeOf<unknown>();
+  });
+
+  it("defaults clientData to unknown when called without a generic", () => {
+    const start = chat.createStartSessionAction("fake-chat");
+    expectTypeOf(start).parameter(0).toHaveProperty("clientData");
+    // Untyped variant — clientData is `unknown`.
+    expectTypeOf<Parameters<typeof start>[0]["clientData"]>().toEqualTypeOf<unknown>();
+  });
```
[the remark this answers is not in the document]
Comment (ericallam) [reply packages/trigger-sdk/src/v3/createStartSessionAction.test.ts:135]: Good catch on the edge case — verified the assertion holds. Typecheck is green in CI ([latest run](https://github.com/triggerdotdev/trigger.dev/actions/runs/26219230552)) and `pnpm run typecheck` + the test pass locally.

When `TChat` defaults to `AnyTask = Task<string, any, any>`, TypeScript's `infer TMetadata` extraction from `ChatTaskWirePayload<any, infer TMetadata>` resolves to `unknown` (not `any`) in this conditional shape — that's why `Parameters<typeof start>[0]["clientData"]` correctly comes out as `unknown` and the `toEqualTypeOf<unknown>()` assertion passes.

**Review by d-cs — APPROVED** [review]
